src-tauri/src/repo/messages.rs

```diff
@@ -261,6 +261,32 @@ where
     .with_context(|| "Failed to get last message")?)
 }
 
+/// Counting messages from Assistant role that are not is_internal_tool_output.
+///
+/// # Errors
+///
+/// Returns error if there was a problem while counting of messages.
+pub async fn get_count_of_failed_messages<'a, E>(executor: E, chat_id: i64) -> Result<i64>
+where
+    E: Executor<'a, Database = Sqlite>,
+{
+    let result = query!(
+        r#"
+        SELECT count(*) as msg_count
+        FROM messages
+        WHERE chat_id = $1
+        AND role = "Assistant"
+        AND is_internal_tool_output IS FALSE
+        "#,
+        chat_id,
+    )
```

Comment:
Would be better to pass this as a parameter instead of a string value, this way we would rely on Rust types to validate the correctness of this string.

I like to rely on Rust types! :)

```suggestion
    let result = query!(
        r#"
        SELECT count(*) as msg_count
        FROM messages
        WHERE chat_id = $1
        AND role = $2
        AND is_internal_tool_output IS FALSE
        "#,
        chat_id,
        Role::Assistant,
    )
```